Adding Client hostname to Sql server connection

saurabh500 opened the pull request:
While connecting to the Sql Server, the SqlClient was sending an empty hostname to the server. 
In this pull request we use using the Environment.MachineName to populate the hostname while connecting to SqlServer.

Fixes #12627 

cc @stephentoub @corivera

justinvp commented on src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnectionString.cs:
```diff
@@ -521,7 +521,7 @@ internal string ObtainWorkstationId()
                 // since permission to open the connection has been granted
                 // the information is shared with the server, but not directly with the user
-                result = string.Empty;
+                result = ADP.MachineName();
```
Just curious: In .NET Framework, the length of the machine name is validated. Is the length check not needed for .NET Core?

https://github.com/Microsoft/referencesource/blob/master/System.Data/System/Data/SqlClient/SqlConnectionString.cs#L701

saurabh500 commented:
> Is there a test we should add?

@stephentoub  Yes. I will be adding another PR with the test.

saurabh500 commented:
@stephentoub The build for Innerloop Linux ARM Emulator SoftFP Debug Cross...  is failing. What is the full title of this step? I want the bot to execute this step.

saurabh500 commented:
@dotnet-bot Test Innerloop Linux ARM Emulator SoftFP Debug Cross Build and Test please

saurabh500 commented:
@stephentoub I have added the tests to this PR itself.

I have modified the ManualTest to build on 1.7 to access `Environment.MachineName` The other manual tests succeed with the migration.
